module1/homework/make_unique/make_unique.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+namespace cs {
+
+template <typename T>
+using IsNotArrayType = std::enable_if_t<!std::is_array_v<T>>;
+
+template <typename T,
+          typename... Args,
+          typename = IsNotArrayType<T>>
+auto make_unique(Args&&... args) {
+    return std::unique_ptr<T>{new T{std::forward<Args>(args)...}};
+}
+
+template <typename T>
+auto make_unique(size_t size) {
+    return std::unique_ptr<T>{new std::remove_extent_t<T>[size] {}};
+}
```

Comment:
hmm... for me we can have possible memory leaks here. Can you check if valgrind reports something in MakeUniqueArrayTest? I can see that sanitizers are ok.

Reply:
Nope. Valgrind is clean
```
HEAP SUMMARY:
==23882==     in use at exit: 0 bytes in 0 blocks
==23882==   total heap usage: 277 allocs, 277 frees, 119,787 bytes allocated
==23882== 
==23882== All heap blocks were freed -- no leaks are possible
==23882== 
==23882== For lists of detected and suppressed errors, rerun with: -s
==23882== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

Reply:
Yes, I thought that there will be T, but array is automatically deduced (eg. T is int[]), so there is no need to have T[]